cupy/_core/_dtype.pyx

```diff
@@ -124,3 +124,28 @@ cpdef void _raise_if_invalid_cast(
     raise TypeError(
         f'Cannot cast {argname} from {from_dt!r} to {to_dt!r} '
         f'according to the rule \'{casting}\'')
+
+
+cdef dict dtype_format = {
+    intern("?"): intern("?"),
+    intern("b"): intern("b"),
+    intern("h"): intern("h"),
+    intern("i"): intern("i"),
+    intern("l"): intern("l"),
+    intern("q"): intern("q"),
+    intern("B"): intern("B"),
+    intern("H"): intern("H"),
+    intern("I"): intern("I"),
+    intern("L"): intern("L"),
+    intern("Q"): intern("Q"),
+    intern("e"): intern("e"),
+    intern("f"): intern("f"),
+    intern("d"): intern("d"),
+    intern("F"): intern("Zf"),
+    intern("D"): intern("Zd"),
```

Comment:
No, not stupid at all, it appears to be a NumPy thing, but because we're using NumPy dtypes I just followed along.
https://github.com/numpy/numpy/blob/d1724e3a5f37a2860dc9ad2b29f0e5a3501d9848/numpy/_core/src/multiarray/buffer.c#L396-L397
Example:
```python
>>> import numpy as np
>>> a = np.empty(10, dtype=np.complex128)
>>> m = memoryview(a)
>>> m.format
'Zd'
```

Reply:
I actually couldn't find good docs for it, it is here: https://peps.python.org/pep-3118/#additions-to-the-struct-string-syntax
(Sorry, very randomly saw this, but may look at this a bit more some time.  If you think I should take a close look, please give a ping.)

Reply:
Nice, thanks @seberg! I see `Zd` is called out here:
https://peps.python.org/pep-3118/#examples-of-data-format-descriptions